### PR TITLE
prompt=none carried in openid redirect

### LINKED ISSUE #1673: prompt=none not carried for silent authentication
The [documentation](https://auth0.com/docs/api-auth/tutorials/silent-authentication) is more complex that it needs to be. 

To authenticate you visit `/v1/openid/auth0/login?callback=http...&scopes=...` it redirects to the OIDC provider (e.g. `https://peterbecom.auth0.com/authorize?client_id=...&response_type=code&scope=openid+email+profile&redirect_uri=...&state=...`) which then redirects you back to the `callback` (if all goes well with the password check) with a code that kinto trades in for an access token which it then appends to the `callback` URL.

Now, if that access token expires (or is *about to* expire!) we need to authenticate again. The default expiration time for the access token (at least with Auth0) is 1 day. To do that, you can simply add `&prompt=none` to the URL. E.g. `https://peterbecom.auth0.com/authorize?client_id=...&response_type=code&scope=openid+email+profile&redirect_uri=...&state=...&prompt=none`. What the OIDC provider does is...

> "Silent authentication lets you perform an authentication flow where Auth0 will only reply with redirects, and never with a login page."

It's basically the same URL as authorization but with `&prompt=none` added.

### PR DESCRIPTION
**Pull request by peterbe**
Fixes #1673

- [x] Add documentation.
- [x] Add tests.
- [x] Add a changelog entry.

### CODE REVIEW

**Comment by peterbe** [comment]
This solved the problem I had. In my React app I keep an eye on the `expires_at` that gets set in `localStorage`. If my code deems the access token too old, it triggers the same code path as was used to redirect to the kinto authorize endpoint **but** with `&prompt=none` added. 

If I try to go to `http://localhost:8888/v1/openid/auth0/login?callback=http%3A%2F%2Flocalhost%3A3000%2F%23provider%3Dauth0%26tokens%3D&scope=openid%20email%20profile&prompt=none` I get:
```
< HTTP/1.1 307 Temporary Redirect
< Location: https://peterbecom.auth0.com/authorize?client_id=R6R1t40cXQnuPQBdC8D83WOVyuUppZdw&response_type=code&scope=openid+email+profile&redirect_uri=http%3A%2F%2Flocalhost%3A8888%2Fv1%2Fopenid%2Fauth0%2Ftoken%3F&state=de6b3235872078a889f2c9134fe8c65c3167ee6819bf71289473570f8c927b74
```
However, with the patch in this PR, that HTTP response becomes this instead:
```
< HTTP/1.1 307 Temporary Redirect
< Location: https://peterbecom.auth0.com/authorize?client_id=R6R1t40cXQnuPQBdC8D83WOVyuUppZdw&response_type=code&scope=openid+email+profile&redirect_uri=http%3A%2F%2Flocalhost%3A8888%2Fv1%2Fopenid%2Fauth0%2Ftoken%3F&state=de6b3235872078a889f2c9134fe8c65c3167ee6819bf71289473570f8c927b74&prompt=none
```
The only difference is the extra `&prompt=none` at the end of the URL. Auth0 needs that so that when my browser arrives there at `https://peterbecom.auth0.com/authorize...` it doesn't ever present a 200 OK. Instead the only thing that can and should happen is that it redirects back to `http://localhost:8888/v1/openid/auth0/login` with a code or with an error.

**Comment by peterbe** [comment]
Note, I don't know where to add this to the documentation. Tips appreciated.

File: kinto/plugins/openid/views.py [hunk 1]
```diff
@@ -94,6 +96,14 @@ def get_login(request):
             'description': "Provider %s requires 'email' scope" % provider,
         }
         raise_invalid(request, **error_details)
+    if prompt is not None and prompt != 'none':
+        # If the 'prompt' query string parameter is set it better be equal
+        # # to 'none' as that's the only value recognized.
+        error_details = {
+            'name': 'prompt',
+            'description': "prompt argument not recognized (%r)" % prompt,
```
Comment (leplatrem) [inline kinto/plugins/openid/views.py]: there might be a `location` key to set to `querystring` if I'm not wrong

Comment (leplatrem) [reply kinto/plugins/openid/views.py]: and I think you could even check that at the schema level actually

File: CHANGELOG.rst [hunk 2]
```diff
@@ -16,6 +16,8 @@ This document describes changes between each past release.
   ``instance_uri_registry``, suitable for use when you don't
   necessarily have a ``request`` object around. The existing functions
   will remain in place.
+- openid plugin will carry ``prompt=none`` query string parameter if appended
```
Comment (leplatrem) [inline CHANGELOG.rst]: I think we wrote *query string* in one word elsewhere...

File: CHANGELOG.rst [hunk 3]
```diff
@@ -16,6 +16,8 @@ This document describes changes between each past release.
...
+  to authorize endpoint.
```
Comment (leplatrem) [inline CHANGELOG.rst]: If you still have some energy we could mention how to use this in the openid docs section ? :) :)

File: kinto/plugins/openid/views.py (hunk 1 above)
Comment (leplatrem) [reply kinto/plugins/openid/views.py]: something like `String(validator=colander.Regex("none"), missing=colander.drop)`

Comment (peterbe) [reply kinto/plugins/openid/views.py]: `TypeError: __init__() got an unexpected keyword argument 'validator'`

Comment (peterbe) [reply kinto/plugins/openid/views.py]:
```
prompt = colander.SchemaNode(colander.String(),
                                 validator=colander.Regex("none"),
                                 missing=colander.drop)
```
worked

**Comment by peterbe** [comment]
@leplatrem Hopefully all feedback addressed.

**Pull request merged**